client/src/main/java/com/flipkart/flux/client/intercept/TaskInterceptor.java

```diff
@@ -162,7 +162,7 @@ private String generateStateIdentifier(Method method) {
         return method.getName();
     }
 
-    private String generateTaskIdentifier(Method method) {
+    public String generateTaskIdentifier(Method method) {
```

Comment:
use new MethodId(method).toString() wherever you need a task identifier.
Make this private again. 
You have included this class in another class just for this method :) Keep the boundaries clean